scalacheck/src/main/scala/pathy/scalacheck/RandomSeg.scala

```diff
@@ -28,9 +28,10 @@ private[scalacheck] object RandomSeg {
   implicit val randomSegArbitrary: Arbitrary[RandomSeg] =
     Arbitrary {
       Gen.nonEmptyListOf(Gen.frequency(
-        100 -> Arbitrary.arbitrary[Char],
+        100 -> Gen.alphaNumChar,
          10 -> Gen.const('.'),
-         10 -> Gen.const('/')
+         10 -> Gen.const('/'),
+          5 -> Arbitrary.arbitrary[Char]
```

Comment:
Ah, I spoke too soon I guess. This line does mean that we preserve the semantic of a random segment. I could be convinced this change in relative frequencies is for the better.